egsnrc2py/gen_commons.py

```diff
@@ -102,5 +102,37 @@ def make_common_file(filename: str):
         f.write("\n".join(lines))
 
 
+def make_egsfortran_globals(egsfortran_name="egsfortran") -> str:
+    """Produce Python global variables referencing f2py common blocks
+
+    This is for mixing new Python code with old Fortran code
+    while transitioning to a (maybe someday) full Python conversion
```

Comment:
I really like this approach. We can potentially have a part Python/part Fortran code base, with the first goal to make it work appropriately, and then to iterate towards conversion while making sure it keeps working.

Reply:
> We can potentially have a part Python/part Fortran code base, with the first goal to make it work appropriately, and then to iterate towards conversion while making sure it keeps working

Totally agree -- that's exactly what I've been working towards.  After being immersed in the "transpiling" type conversion I realized that ensuring correctness is key, so building this kind of bridge is quite important.  At each stage add tests with the correct outputs (from normal Fortran run), shift the next piece to Python, make sure tests are passing, etc.  Then the tests leave us in a good place for whatever restructuring of the Python code happens later.

Next commit has `tutor1` and `shower` both written in Python, shower just calls the fortran `electr`.  Got all the commons defined, got it executing, but the output is wrong... now trying to determine what is going on.

After `shower` is in place, can replace the `electr` call with the Python code (or more likely parts of `electr`) when ready.  Kind of working from both the front end and the back end (replacing small parts) to try to meet in the middle.

Reply:
Beautiful 🙂